packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt

```diff
@@ -97,6 +98,13 @@ class FlutterPlugin : Plugin<Project> {
             repositories.maven {
                 url = uri(repository!!)
             }
+            if (plugins.hasPlugin("com.android.application") && hasProperty("android.injected.invoked.from.ide")) {
```

Comment:
Where did you find "android.injected.invoked.from.ide"? I would like to read the source/documentation. 

In general we dont do ide specific code in our plugin but from the string I cant tell if this is part of the api or something that is android studio specific.

Reply:
I found it on StackOverflow: https://stackoverflow.com/q/18824890 I just want to inject the dependencies for Android Studio in normal builds like the CLI or CI those dependencies are not relevant.

Reply:
@jwren Is there something equivalent for visual studio or something ide neutral you want this pr to include? 
For now it is ok to leave this because I want @gmackall's opinion but I think we may ask that the check for android studio be removed to align with our philosophy of of shipping tools that are portable. 

It also helps protect against bugs that only happen for users when they run from android studio. This is extra important because we do not have integration tests that require launch android studio and verifying the tooling works.

Reply:
Source appears to be here https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/builder-model/src/main/java/com/android/builder/model/InjectedProperties.kt?q=%20PROPERTY_INVOKED_FROM_IDE

Reply:
If this change has been tested and seems to work then I am okay with it.  Any other IDE neutral changes can be added as a follow on.

@rekire -- Thanks for this change.  Did you check that this doesn't have any negative effects for Ultimate or Community?

Reply:
Not yet, I can try it tomorrow on the CE, but I've no license for the ultimate edition.

Reply:
CE is sufficient

Reply:
I tested it with my new template and it worked as expected. I just had to click once on import gradle project or something similar which popped up on the right. I guess that is the default behavior. Visual Studio code seems not to break ether, however there I had no Kotlin support (most likely because I never installed those plugins)